transcom/mymove
# B 20639 nts loa fetching int

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20639)

## Summary
This ticket was rewritten after some NTS review. We're now introducing a new column to the move table to explicitly track when a move task order gets approved. Currently, the "MTO approved" timestamp is the `available_to_prime_at` database value, however in the future third party vendors may be available and NTS shipments will need a specific attribute just for MTO approval dates in order to look up the LOAs.

We are also retroactively applying this logic by setting the `approved_at` value on all existing MTOs to be equal to the `available_to_prime_at` value.

### How to test

1. Login as a new customer and create a new HHG move
2. Login as service counselor, review the order, and submit the move details
3. Login as task ordering officer and approve the requested move
4. Open your browser tools network tab -> refresh the page
5. Look for your move locator within the network tab, once found open it and verify a request URL of `ghc/v1/move/4C3XQM`
6. Read the response and verify that the attribute `approvedAt` is present and holds the same value as `availableToPrimeAt

### Database

#### Any new migrations/schema changes:

- [x] Excel sheet was updated

Zero-downtime not necessary as this does not affect existing columns
